packages/components/inputs/date-input/src/date-input.spec.js

```diff
@@ -221,3 +226,32 @@ describe('date picker keyboard navigation', () => {
     });
   });
 });
+
+it('should only emit valid dates from manually entered datestrings', async () => {
+  // Render the input with an initial value
+  renderDateInput({ value: '2020-09-15', 'data-testid': 'onblurtest' });
+  const htmlInputElement = screen.getByTestId('onblurtest');
+
+  // verify it got formatted for display
+  await waitFor(() =>
+    expect(htmlInputElement).toHaveDisplayValue('09/15/2020')
+  );
+
+  // enter a valid formatted date
+  await fireEvent.change(htmlInputElement, { target: { value: '03/28/2024' } });
+  await fireEvent.blur(htmlInputElement);
+
+  // no change is expected
+  await waitFor(() =>
+    expect(htmlInputElement).toHaveDisplayValue('03/28/2024')
+  );
+
+  // enter an invalid date
+  await fireEvent.change(htmlInputElement, { target: { value: '33/28/2024' } });
+  await fireEvent.blur(htmlInputElement);
+
+  // should reset to the most recent valid date
```

Comment:
nit: This is the only part I'm unclear about - is it better to return the most recent valid date, or for validation to fail and the input to go into an error state?

If this is the behavior for typed values elsewhere, then it's outside of the scope of this PR and this comment should be closed.

Reply:
Right now, there’s no error handling in place. Before, users couldn’t select an invalid date through the calendar UI. If they entered an invalid date and pressed 'return,' the calendar wouldn't close, and no value would be passed along.

Now, with the possibility that an invalid date could be entered, I wanted to make sure that this incorrect value doesn’t get passed through, as some consumers might assume this input always provides a valid date. The only other option I see would be to clear the input entirely, but that wouldn’t allow users to easily correct small mistakes, like typos.

Reply:
that works for me